### PR TITLE
Create Page model

### PR DESCRIPTION
**Pull request by vbrown608**
@wioux what do you think of this direction for creating a page model to manage generic content? I wasn't totally sure what you had in mind with #188.

I put a body field on the page model, though we could just use ManagedContent throughout. I figured typical pages will have a body, and rarer special cases may also have some ManagedContent instances attached. I updated `credits` and `materials-overview` so far.

For now I just renamed the Page class within `articles-overview`. That should get rolled in eventually.

### CODE REVIEW

**Comment by wioux** [comment]
This is good. The only thing I would suggest is we could register pages with activeadmin and then delete `{blog,materials,topics}_overview.rb`, and instead replace those links in the admin menu with ones that go directly to the right edit form.

**Comment by vbrown608** [comment]
Great, thanks for the feedback.

**Comment by vbrown608** [comment]
Okay, I did some more work on this. I have a couple open questions, input much appreciated.
- In https://github.com/EFForg/sec/pull/246/commits/e3f236fce9e153a2f85b27273341ee4199b32475 I tried to update the ArticlesPage class to inherit from Page. It causes an error on subsequent reloads after starting the dev server. I can try to sort it out of this seems like a reasonable refactor, but I wanted to check in about the approach.
- a Page `has_many :manged_content`, but none of the pages actually have any at the moment. I imagined  attaching it to pages when more bits of content are needed, like on the homepage (and maybe renaming to snippet) but I could also just pull it out. I'm not too keen to redo the homepage right now, was thinking I'd leave it as a singleton unless we come upon a good reason to convert it to a Page.
- Setting up the pages with their names in the seeds/migration and then expecting them to be there seems a little flimsy to me. I wonder if it would be better to store an array of page names that are expected to exist on the Page model and create them on attempted retrieval if they don't exist.

**Comment by wioux** [comment]
The first two migrations fail because the slug column doesn't exist yet but the friendly_id config has already been set up and expects it to. I worked around it by adding the slug column in the original CreatePage migration.

**Comment by wioux** [comment]
> In e3f236f I tried to update the ArticlesPage class to inherit from Page. It causes an error on subsequent reloads after starting the dev server. I can try to sort it out of this seems like a reasonable refactor, but I wanted to check in about the approach.

That does seem reasonable. I think reloading will work better if the ArticlesPage class is moved into an autoloaded directory like app/models.

**Comment by wioux** [comment]
> a Page has_many :manged_content, but none of the pages actually have any at the moment. I imagined attaching it to pages when more bits of content are needed, like on the homepage (and maybe renaming to snippet) but I could also just pull it out. I'm not too keen to redo the homepage right now, was thinking I'd leave it as a singleton unless we come upon a good reason to convert it to a Page.

I think it being unused and poorly named is a good reason to yank it if you want. The homepage appears to be a very special case.

**Comment by wioux** [comment]
This removed a lot duplication in the admin area, nice! :+1: It should also let us clean up the Articles Overview admin page in a follow up, now that we have a parent model nest article sections under. The homepage issues I was originally thinking about don't seem pressing at this point so I think we can say this closes #188. I'll make a separate issue for Articles Overview.

**Comment by wioux** [comment]
:100: ready to go once the merge conflict is resolved.

**Comment by vbrown608** [comment]
I think the removal of "managed content" is messing up the migration - I'm going to take that association out the original creation of the page model.

File: app/views/credits/index.html.erb [hunk 1]
```diff
@@ -2,7 +2,6 @@
   <h1>Credits</h1>
 
   <div class="content">
-    <%= managed_content("credits") %>
+    <% html(@page.body) %>
```
Comment (wioux) [inline app/views/credits/index.html.erb]: I think this was working before #440 was merged, but now we need an equals here (ie `<%= html(@page.body) %>`).

**Pull request merged**